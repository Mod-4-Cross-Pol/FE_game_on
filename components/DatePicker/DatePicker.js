import React, { useState } from 'react';
import DateTimePicker from '@react-native-community/datetimepicker';
import { StyleSheet, View, Text, Platform, Button, Alert } from 'react-native';
import { connect } from 'react-redux';
import { saveSelectedDate } from '../../actions/actions';

export function DatePicker(props) {

  const [ date, setDate ] = useState(Date.now());
  const [ mode, setMode ] = useState('date');
  const [ show, setShow ] = useState(false);

  const onChange = async (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShow(Platform.OS === 'ios');
    setDate(currentDate);
    let displayDate = selectedDate.toString()
    displayDate = displayDate.split(' ').splice(0,4).join(' ')
    await props.saveSelectedDate(displayDate)
  };

  return (
    <View style={styles.container}>
      <View style={styles.textBox}>
        <Text style={styles.text}>SELECT A DAY</Text>
      </View>
      <DateTimePicker
        style={styles.picker}
        testID="dateTimePicker"
        value={date}
        mode={mode}
        display="calender"
        onChange={onChange}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
    backgroundColor: '#28BFBD'
  },
  textBox: {
    height: 200,
    paddingBottom: 10,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    fontSize: 30,
    color: '#FFF',
    fontWeight: 'bold'
  },
  picker: {
    marginTop: -100,
    height: 500
  }
});

export const mapStateToProps = (state) => ({
  selectedDate: state.selectedDate
});

export const mapDispatchToProps = (dispatch) => ({
  saveSelectedDate: targetDate => dispatch(saveSelectedDate(targetDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(DatePicker);
