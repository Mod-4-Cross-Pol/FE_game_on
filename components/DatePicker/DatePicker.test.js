import React from 'react';
import { DatePicker } from './DatePicker';
// import DateTimePicker from '@react-native-community/datetimepicker';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('DatePicker', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = Enzyme.shallow(<DatePicker />);
  })

  test('verfy testing file works', () => {
    expect(true).toBeTruthy();
  });

  test('verfy testing file works', () => {
    expect(wrapper).toMatchSnapshot();
  });

});






// import React from 'react';
// import { DatePicker } from './DatePicker';
// import DateTimePicker from '@react-native-community/datetimepicker';
// import renderer from 'react-test-renderer';
//
// describe('DatePicker', () => {
//
//   describe('Renders a DatePicker component', () => {
//
//     test('renders correctly', () => {
//       const wrapper = renderer.create(<DatePicker />).toJSON();
//       expect(wrapper).toMatchSnapshot();
//     });
//
//
//   });
//
// });
