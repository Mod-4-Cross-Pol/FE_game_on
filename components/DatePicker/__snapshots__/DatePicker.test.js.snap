// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker verfy testing file works 1`] = `ShallowWrapper {}`;
